File: 0386 Lexicographical Numbers/0386.py
class Solution:
    def lexicalOrder(self, n: int) -> List[int]:
        ans = []
        num = 1

        for _ in range(n):
            ans.append(num)
            if num * 10 <= n:
                num *= 10
            else:
                while num % 10 == 9 or num + 1 > n:
                    num //= 10
                num += 1

        return ans

File: 0386 Lexicographical Numbers/0386.cpp
class Solution {
public:
    vector<int> lexicalOrder(int n) {
        vector<int> ans;
        int num = 1;
        int left = n;

        while (left--){
            ans.push_back(num);
            if (num * 10 <= n){
                num *= 10;
            } else {
                while (num % 10 == 9 || num + 1 > n){
                    num /= 10;
                }
                num++;
            }
        }

        return ans;
    }
};
